### PR TITLE
change service path and update list on ui 

### PR DESCRIPTION
**Pull request by shayb**
update list on ui thread
add default package for services

### CODE REVIEW

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt [hunk 1]
```diff
@@ -25,15 +28,21 @@ class AppsDiscoveryActivity : AppCompatActivity() {
         loader = findViewById(R.id.loader)
         findViewById<AppsDiscoveryList>(R.id.list).setLoadingListener(object : AppsDiscoveryListPresenter.LoadingListener {
             override fun loading() {
-                updateLoading()
+                handler.post {
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt]: this shouldn't be done here. Make sure that in `DiscoverAppsRepository` where you execute the background thread, you always respond using the ui thread and then you won't need to wrap with a handler here.

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt [hunk 2]
```diff
@@ -25,15 +28,21 @@ class AppsDiscoveryActivity : AppCompatActivity() {
...
+                    updateLoading()
+                }
             }
 
             override fun loadingComplete() {
-                updateLoadingComplete()
+                handler.post {
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt]: same

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt [hunk 3]
```diff
@@ -25,15 +28,21 @@ class AppsDiscoveryActivity : AppCompatActivity() {
...
+                    updateLoadingComplete()
+                }
             }
 
             override fun loadingFailed() {
-                updateLoadingFailed()
+                handler.post {
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt]: same

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/customView/AppsDiscoveryList.kt [hunk 4]
```diff
@@ -33,7 +33,7 @@ class AppsDiscoveryList @JvmOverloads constructor(context: Context, attrs: Attri
     init {
         layoutManager = GridLayoutManager(context, COLUMNS_COUNT)
         val discoveryAppsRepository = DiscoveryAppsRepository.getInstance(context.packageName, DiscoveryAppsLocal(context), DiscoveryAppsRemote(), Handler(Looper.getMainLooper()))
-        presenter = AppsDiscoveryListPresenter(discoveryAppsRepository)
+        presenter = AppsDiscoveryListPresenter(discoveryAppsRepository, Handler(Looper.getMainLooper()))
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/customView/AppsDiscoveryList.kt]: no need

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/repositories/DiscoveryAppsRepository.kt [hunk 5]
```diff
@@ -73,15 +73,13 @@ private constructor(private val currentPackage: String, private val local: Disco
...
-                        var filterApps: List<EcosystemApp> = listOf()
+                        var filterApps: List<EcosystemApp> = serverApps
                         val currentApp: EcosystemApp? = serverApps.firstOrNull { it.identifier == currentPackage }
+                        //if finds this app in the list - remove it from the list
                         currentApp?.let { app ->
                             local.appIconUrl = app.iconUrl
                             local.memo = app.memo
                             filterApps = serverApps.filter { it != app }
-                        } ?: kotlin.run {
-                            listener.onError("cant find in apps the current package $currentPackage")
```
Comment (berryve) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/repositories/DiscoveryAppsRepository.kt:83]: the problem was that you were not running this ^^ on the UI thread

Comment (berryve) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/repositories/DiscoveryAppsRepository.kt:83]: if you look a few lines below in this method when there is success you do:
```
uiHandler.post {
  listener.onResult(discoveryApps)
}
```
So the success onResult call is always called on the ui thread.
This explains why the regular success + regular error messages worked fine on the stage sample app.

**Comment by shayb** [comment]
fixed issues

**Pull request merged**